Corgibytes.Freshli.Cli/Functionality/Git/GitSource.cs

```diff
@@ -149,14 +149,47 @@ private void Checkout(string gitPath)
 
     private void Pull(string gitPath)
     {
+        var stdOutBuffer = new StringBuilder();
+        var branch = Branch;
+        if (Branch == null)
+        {
+            branch = FetchCurrentBranch(gitPath);
+        }
+
         var stdErrBuffer = new StringBuilder();
         var command = CliWrap.Cli.Wrap(gitPath).WithArguments(
                 args => args
                     .Add("pull")
                     .Add("origin")
-                    .Add(Branch ?? "")
+                    .Add(branch ?? "")
```

Comment:
If `""` is passed here, this command will fail. We should raise an exception if `branch` is null or empty by this point.

Reply:
It won't fail actually. The branch is optional when pulling, it'll fallback to the default branch. 

To be certain, I've tested it with `.Add("")` instead of `.Add(branch ?? "")` and I didn't get any exceptions.